sdk/core/azure-core-amqp/src/main/java/com/azure/core/amqp/models/AmqpMessageBody.java

```diff
@@ -3,14 +3,115 @@
 
 package com.azure.core.amqp.models;
 
+import com.azure.core.util.IterableStream;
+import com.azure.core.util.logging.ClientLogger;
+
+import java.util.Collections;
+import java.util.List;
+import java.util.Objects;
+
 /**
- * Interface representing Amqp Message Body.
+ * This class encapsulates the body of a message. The {@link AmqpMessageBodyType} map to an AMQP specification message
+ * body types. Current implementation only support {@link AmqpMessageBodyType#DATA DATA} AMQP data type. Track this
+ * <a href="https://github.com/Azure/azure-sdk-for-java/issues/17614" target="_blank">issue</a> to find out support for
+ * other AMQP types.
+ *
+ * <p><b>Client should test for {@link AmqpMessageBodyType} before calling corresponding get method. Get methods not
+ * corresponding to the type of the body throws exception.</b></p>
+ *
+ * <p><strong>How to check for {@link AmqpMessageBodyType}</strong></p>
+ * {@codesnippet com.azure.core.amqp.models.AmqpBodyType.checkBodyType}
+ *
+ * @see AmqpMessageBodyType
  */
-public interface AmqpMessageBody {
+public final class AmqpMessageBody {
+    private final ClientLogger logger = new ClientLogger(AmqpMessageBody.class);
+    private AmqpMessageBodyType bodyType;
+
+    // We expect user to call `getFirstData()` more because we support one byte[] as present.
+    // This the priority here to store payload as `byte[] data` and
+    private byte[] data;
+    private List<byte[]> dataList;
+
+    private AmqpMessageBody() {
+        // package constructor so no one can create instance of this except classes im this package.
```

Comment:
nit: Update comment as this is not a package-private constructor anymore or remove the comment